lib/include/openamp/remoteproc_virtio.h

```diff
@@ -99,26 +107,6 @@ rproc_virtio_create_vdev(unsigned int role, unsigned int notifyid,
  */
 void rproc_virtio_remove_vdev(struct virtio_device *vdev);
 
-/**
- * rproc_virtio_create_vring
- *
- * Create rproc virtio vring
- *
- * @vdev: pointer to the virtio device
- * @index: vring index in the virtio device
- * @notifyid: remoteproc vring notification id
- * @va: vring virtual address
- * @io: pointer to vring I/O region
- * @num_desc: number of descriptors
- * @align: vring alignment
- *
- * return 0 for success, negative value for failure.
- */
-int rproc_virtio_init_vring(struct virtio_device *vdev, unsigned int index,
-			    unsigned int notifyid, void *va,
-			    struct metal_io_region *io,
-			    unsigned int num_descs, unsigned int align);
-
```

Comment:
This is one option I was thinking too. I just think it  better not to have busy loop in the library. I was also thinking about the one in the RPMsg side.
Also the way I did it is because previously i also did vring init which will get the da addr from resource table from remoteproc_create_virtio(). I prefer to have this function returns virtio device but not wait for remote to make status OK as how it behaves now to have more flexibility. And want to wrap the details on setting up the vrings in the library. And the remoteproc_virtio is more like a internal structure to remoteproc, user only directly use remoteproc and calls remoteproc APIs instead of remoteproc_virito APIs and thus, i think it can be fine to change the internal remoteproc_virtio.

Reply:
>This is one option I was thinking too. I just think it better not to have busy loop in the library. I was also thinking >about the one in the RPMsg side.
>Also the way I did it is because previously i also did vring init which will get the da addr from resource table from >remoteproc_create_virtio(). I prefer to have this function returns virtio device but not wait for remote to make status >OK as how it behaves now to have more flexibility. 
Agree with that, this was a point i mentioned when i have proposed the patch . From my pov we need a non blocking API.

>And the remoteproc_virtio is more like a internal structure to remoteproc, user only directly use remoteproc and calls >remoteproc APIs instead of remoteproc_virito APIs and thus, i think it can be fine to change the internal >remoteproc_virtio.
Not in line with this. We discussed the possibility to have a virtio backend based on resource table that does not dependent on remoteproc framework. The remoteproc virtio was the solution you proposed, and it is the solution we implement now in ST projects. Now i you decide to reintegrate the code in remoteproc, my concern is that there is no more backend available for user decoupled from the remoteproc, because we need to use the rproc to initialize the rpmsg.